File: src/main/java/fr/spoutnik87/musicbot_rest/repository/UserRepository.java
package fr.spoutnik87.musicbot_rest.repository;

import fr.spoutnik87.musicbot_rest.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

  Optional<User> findByEmail(String email);

  Optional<User> findByNickname(String nickname);

  Optional<User> findByUuid(String uuid);
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/repository/UserRepository.kt
package fr.spoutnik87.musicbot_rest.repository

import fr.spoutnik87.musicbot_rest.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
interface UserRepository : JpaRepository<User, Long> {
    @Transactional(readOnly = true)
    fun findByUuid(uuid: String): User?
    @Transactional(readOnly = true)
    fun findByEmail(email: String): User?
    @Transactional(readOnly = true)
    fun findByNickname(nickname: String): User?
    @Transactional(readOnly = true)
    fun findByUserId(userId: String): User?
}